awx/sso/conf.py

```diff
@@ -842,6 +842,292 @@ def _register_ldap(append=None):
+    category=_('GitHub OAuth2'),
+    category_slug='github-enterprise',
+    encrypted=True,
+)
+
+register(
+    'SOCIAL_AUTH_GITHUB_ENTERPRISE_ORGANIZATION_MAP',
+    field_class=SocialOrganizationMapField,
+    allow_null=True,
+    default=None,
+    label=_('GitHub Enterprise OAuth2 Organization Map'),
+    help_text=SOCIAL_AUTH_ORGANIZATION_MAP_HELP_TEXT,
+    category=_('GitHub Enterprise OAuth2'),
+    category_slug='github-enterprise',
+    placeholder=SOCIAL_AUTH_ORGANIZATION_MAP_PLACEHOLDER,
+)
+
+register(
+    'SOCIAL_AUTH_GITHUB_ENTERPRISE_TEAM_MAP',
+    field_class=SocialTeamMapField,
+    allow_null=True,
+    default=None,
+    label=_('GitHub Enterprise OAuth2 Team Map'),
+    help_text=SOCIAL_AUTH_TEAM_MAP_HELP_TEXT,
+    category=_('GitHub Enterprise OAuth2'),
+    category_slug='github-enterprise',
+    placeholder=SOCIAL_AUTH_TEAM_MAP_PLACEHOLDER,
+)
+
+###############################################################################
+# GITHUB ENTERPRISE ORG OAUTH2 AUTHENTICATION SETTINGS
+###############################################################################
+
+register(
+    'SOCIAL_AUTH_GITHUB_ENTERPRISE_ORG_CALLBACK_URL',
+    field_class=fields.CharField,
+    read_only=True,
+    default=SocialAuthCallbackURL('github-enterprise-org'),
+    label=_('GitHub Enterprise Organization OAuth2 Callback URL'),
+    help_text=_('Provide this URL as the callback URL for your application as part '
+                'of your registration process. Refer to the Ansible Tower '
+                'documentation for more detail.'),
+    category=_('GitHub Enterprise Organization OAuth2'),
+    category_slug='github-enterprise-org',
+    depends_on=['TOWER_URL_BASE'],
+)
+
+register(
+    'SOCIAL_AUTH_GITHUB_ENTERPRISE_ORG_URL',
+    field_class=fields.CharField,
+    allow_blank=False,
+    default='',
+    label=_('GitHub Enterprise Organization URL'),
+    help_text=_('The URL for your Github Enteprise.'),
+    category=_('GitHub Enterprise OAuth2'),
+    category_slug='github-enterprise-org',
+)
+
+register(
+    'SOCIAL_AUTH_GITHUB_ENTERPRISE_ORG_API_URL',
+    field_class=fields.CharField,
+    allow_blank=False,
+    default='',
+    label=_('GitHub Enterprise Organization API URL'),
+    help_text=_('The API URL for your GitHub Enterprise.'),
+    category=_('GitHub Enterprise OAuth2'),
+    category_slug='github-enterprise-org',
+)
+
+register(
+    'SOCIAL_AUTH_GITHUB_ENTERPRISE_ORG_KEY',
+    field_class=fields.CharField,
+    allow_blank=True,
+    default='',
+    label=_('GitHub Enterprise Organization OAuth2 Key'),
+    help_text=_('The OAuth2 key (Client ID) from your GitHub Enterprise organization application.'),
+    category=_('GitHub Enterprise Organization OAuth2'),
+    category_slug='github-enterprise-org',
+)
+
+register(
+    'SOCIAL_AUTH_GITHUB_ENTERPRISE_ORG_SECRET',
+    field_class=fields.CharField,
+    allow_blank=True,
```

Comment:
I believe we can remove this 
```
 allow_blank=True,
```

It will complain if this is ever an empty string.

Reply:
Feel free to use this if you want.  https://github.com/ansible/awx/commit/4692e2fd70b3e5d17d10c927c28a57a8f7f39ede

Reply:
I think this is slightly bigger change. I've looked at other fields, and they have same construct (basically this part of PR is just a mirror of existing Gtihub fields with name/description modifications), e.g.:
https://github.com/ansible/awx/blob/3a467067f3705c3a2dd58584fc638266434c2b68/awx/sso/conf.py#L654-L664

If this indeed has to be changed, I think a separate PR would be more suitable, as I believe other fields (not just Github one) might have same "problem".

Reply:
@egmar without changing `allow_blank` to True for these fields, I am seeing these warnings in the awx logs.  

![image](https://user-images.githubusercontent.com/11698892/105086677-5b9bca80-5a67-11eb-9341-5de7f3df906f.png)

Reply:
Hey @egmar, you're right that we want the `allow_blank` content field to stay in there. @rooftopcellist 's edits in https://github.com/ansible/awx/commit/4692e2fd70b3e5d17d10c927c28a57a8f7f39ede will not remove the `allow_blank` field but instead sets them all to `True`, which is what we want. The API will error out otherwise, since not everyone has values to put into these fields, and on newly spun-up AWX instances there are no values, so these fields would give an error on startup unless they're changed to `True`.